addons/product_images/models/ir_cron_trigger.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ir_cron_trigger(models.Model):
+    _inherit = "ir.cron.trigger"
+
+    @api.constrains('cron_id')
+    def _check_cron_id(self):
+        """ Check if `ir_cron_fetch_image` is already triggered
+
+        Ensure that there is a maximum of one trigger at a time for `ir_cron_fetch_image`.
+
+        :return: None
+        :raises ValidationError: If there is already an `ir_cron_fetch_image` triggered
+        """
+        ir_cron_fetch_image_id = self.env.ref('product_images.ir_cron_fetch_image').id
+        if self.cron_id.id != ir_cron_fetch_image_id:
+            return
```

Comment:
Re-reading the code actually I would prefere a defensive programming style like anv's suggestion:

```suggestion
        fetch_image_cron = self.env.ref('product_images.ir_cron_fetch_image', raise_if_not_found=False)
        if fetch_image_cron is None or self.cron_id.id != ir_cron_fetch_image_id:
            return
```

```suggestion
        ir_cron_fetch_image_id = self.env.ref('product_images.ir_cron_fetch_image').id
        if self.cron_id.id != ir_cron_fetch_image_id:
            return
```